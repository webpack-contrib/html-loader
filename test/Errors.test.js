/* eslint-disable */
import loader from '../src';
import webpack from './helpers/compiler';

describe('Errors', () => {
  test('Loader Error', async () => {
    const config = {
      loader: {
        test: /\.html$/,
        options: {
          minimize: true
        }
      }
    };

    const stats = await webpack('error.js', config);
    const { source } = stats.toJson().modules[1];

    // eslint-disable-next-line
    const err = () => eval(source);

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });

  test('Validation Error', () => {
    const err = () => loader.call({ query: { template: 1 } });

    expect(err).toThrow();
    expect(err).toThrowErrorMatchingSnapshot();
  });
});
