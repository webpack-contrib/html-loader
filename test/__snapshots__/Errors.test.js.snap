// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Loader Error 1`] = `
"Module build failed: HTML Loader: 

tree.forEach is not a function
"
`;

exports[`Errors Validation Error 1`] = `
"HTML Loader Invalid Options

options.template should be boolean,string
"
`;
